common-arcana.lic

```diff
@@ -179,4 +186,47 @@ module DRCA
       .map { |_name, properties| properties['abbrev'] }
       .each { |abbrev| fput("release #{abbrev}") }
   end
+  
+  def parse_mana_message(mana_msg)
+      manalevels = nil
+      if mana_msg.include? 'weak'
+        manalevels = $MANA_MAP['weak']
+      elsif mana_msg.include? 'developing'
+        manalevels = $MANA_MAP['developing']
+      elsif mana_msg.include? 'improving'
+        manalevels = $MANA_MAP['improving']
+      else
+        manalevels = $MANA_MAP['good']
+      end
+        
+      adj = mana_msg.split(' ')[-1]
+        
+      return manalevels.index(adj).to_i + 1
+  
+  end
+  
+  def perc_mana
+  
+    return nil if DRStats.barbarian? || DRStats.thief? || DRStats.trader? || DRStats.commoner?
+
+    if DRStats.moon_mage?
+      DRC.bput('perc mana','the Psychic Projection book.')
+      mana_msgs = reget(5)[0..3]
+      
+      mana_msgs.collect! { |mana_msg|
+        mana_msg.split(' streams')[0]
+      }
+      
+      mana_levels = Hash.new
+      mana_levels['enlightened_geometry'] = parse_mana_message(mana_msgs[0])
+      mana_levels['moonlight_manipulation'] = parse_mana_message(mana_msgs[1])
+      mana_levels['perception'] = parse_mana_message(mana_msgs[2])
+      mana_levels['psychic_projection'] = parse_mana_message(mana_msgs[3])
+      return mana_levels
+    else    
+      mana_msg = DRC.bput('perc','^You reach out with your .* and (see|hear) \w+')
+      return parse_mana_message(mana_msg)
```

Comment:
I *suspect* we'll find it easier to use this helper if it returns the same data type for every guild (e.g. a hash with one pair for non-moon mages), but we'll know more once we use it somewhere.